components/prize_pool/commons/prize_pool_import.lua

```diff
@@ -0,0 +1,513 @@
+		priorPlacement.placeEnd or 0
+	)
+end
+
+function Import._mergePlacement(lpdbEntries, placement)
+	for opponentIndex, opponent in ipairs(lpdbEntries) do
+		placement.opponents[opponentIndex] = Import._mergeEntry(
+			opponent,
+			Table.mergeInto(placement:_parseOpponents{{}}[1], placement.opponents[opponentIndex]),
+			placement
+		)
+	end
+
+	assert(
+		#placement.opponents <= 1 + placement.placeEnd - placement.placeStart,
+		'Import: Too many opponents returned from query for placement range '
+			.. placement:_displayPlace():gsub('&#045;', '-')
+	)
+
+	return placement
+end
+
+function Import._mergeEntry(lpdbEntry, entry, placement)
+	if
+		not Opponent.isTbd(entry.opponentData) -- valid manual input
+		or Table.isEmpty(lpdbEntry.opponent) -- irrelevant lpdbEntry
+		or Opponent.isTbd(lpdbEntry.opponent)
+	then
+		return entry
+	end
+
+	entry.opponentData = Import._removeTbdIdentifiers(entry.opponentData)
+	return Table.deepMergeInto(Import._entryToOpponent(lpdbEntry, placement), entry)
+end
+
+function Import._removeTbdIdentifiers(opponent)
+	if Table.isEmpty(opponent) then
+		return {}
+	elseif not Opponent.isTbd(opponent) then
+		opponent.isAlreadyParsed = true
+		return opponent
+	elseif not opponent.type or not Opponent.typeIsParty(opponent.type) then
+		return {}
+	end
+
+	-- Entry is a TBD Opponent, but could contain data (e.g. flag or faction) for party opponents
+	opponent.isAlreadyParsed = true
+	opponent.type = nil
+	for _, player in pairs(opponent.players or {}) do
+		if Opponent.playerIsTbd(player) then
+			player.displayName = nil
+			player.pageIsResolved = nil
+			player.pageName = nil
+		end
+	end
+
+	return opponent
+end
+
+function Import._entryToOpponent(lpdbEntry, placement)
+	local additionalData = {}
+	if lpdbEntry.needsLastVs then
+		additionalData = Import._groupLastVsAdditionalData(lpdbEntry)
+	end
+
+	local score = additionalData.score or Import._getScore(lpdbEntry.opponent)
+	local vsScore = additionalData.vsScore or Import._getScore(lpdbEntry.vsOpponent)
+	local lastVsScore
+	if score or vsScore then
+		lastVsScore = (score or '') .. '-' .. (vsScore or '')
+	end
+
+	return placement:_parseOpponents{{
+		Import._removeTbdIdentifiers(lpdbEntry.opponent),
+		wdl = (not lpdbEntry.needsLastVs) and Import._formatGroupScore(lpdbEntry) or nil,
+		lastvs = {additionalData.lastVs or Import._removeTbdIdentifiers(lpdbEntry.vsOpponent)},
+		lastvsscore = additionalData.score or lastVsScore,
+		date = lpdbEntry.date,
+	}}[1]
+end
+
+function Import._formatGroupScore(lpdbEntry)
+	if not lpdbEntry.scoreBoard then
+		return
+	end
+
+	local matches = lpdbEntry.scoreBoard.match
+	local overtime = lpdbEntry.scoreBoard.overtime
+	local wdl
+	if (matches.d or 0) ~= 0 then
+		wdl = {matches.w or '', matches.d or '', matches.l or ''}
+	elseif (overtime.w or 0) ~= 0 or (overtime.l or 0) ~= 0 then
+		wdl = {matches.w or '', overtime.w or '', overtime.l or '', matches.l or ''}
+	else
+		wdl = {matches.w or '', matches.l or ''}
+	end
```

Comment:
This logic doesn't look correct to me
In a scenario with a tournament with draws, but a specific team got no draws, only W/L will be saved?

Reply:
will have to change so e stuff in current module and the tournament/utils one for it

Reply:
```suggestion
	if String.isNotEmpty(matches.d) then
		wdl = {matches.w or '', matches.d or '', matches.l or ''}
	elseif (overtime.w or 0) ~= 0 or (overtime.l or 0) ~= 0 then
		wdl = {matches.w or '', overtime.w or '', overtime.l or '', matches.l or ''}
	else
		wdl = {matches.w or '', matches.l or ''}
	end
```

Reply:
Same also applies to Overtime

Reply:
and here we get an issue with how we currently store overtime
![EC681C11-3D4A-47A9-87A3-A00BD1242290](https://user-images.githubusercontent.com/75081997/191589588-e682a7bb-1bf4-4076-91cb-f032a25351a6.jpeg)
this is from a gtl from sc2 using the commons storage (with the sc/sc2 wrapper)
we have no overtime and still we store it as 0-0-0 (commons!)
so either we adjust overtime storage in case of no overtime or we have to come up with another solution for this

Reply:
Yes. A team can finish all their match in regulation eg Heroic here
 https://liquipedia.net/rainbowsix/European_League/2022/Stage_1#Standings

Reply:
alright should be good now